### DISCUSSION TITLE
oauth2 multiple routes

### DISCUSSION THREAD

**Discussion by lukasa1993** [Kong/kong#7280] [Ideas and feature requests]
how i enable oauth2 on two routes so i can use 1 access token for both.
i tried adding them separately with same `scope` but it says `unauthorized` any other way achieving this?

**Comment by jeremyjpj0916** [comment]
Did you enable the global config property on both OAUTH2 token plugins?

**Comment by lukasa1993** [comment]
uh no missed that, will it honor scope ?

**Comment by jeremyjpj0916** [comment]
Not sure, just a community user here. But I do know without the ```global``` configuration property enabled, you will not be able to use a token generated from one endpoint by a consumer on another proxy endpoint.

**Comment by lukasa1993** [comment]
doesn't work with global either :(

**Comment by bungle** [comment]
@lukasa1993 is there possibility to map both routes to same service and apply Oauth on service?

**Comment by lukasa1993** [comment]
@bungle no they are actually separate services

**Comment by lukasa1993** [comment]
i cant figure out how to use scopes without this

**Comment by lukasa1993** [comment]
any updates?

**Comment by lukasa1993** [comment]
any hope for this?

**Comment by lukasa1993** [comment]
any love for this ?

**Comment by jeremyjpj0916** [comment]
“ how i enable oauth2 on two routes so i can use the same token.”

I do this in Kong already. I don’t use the scopes field as its not really required for Oauth2 client credential grant pattern. Maybe it would help to post your Kong resources to see where you are going wrong. Setup a token gen route. Make it client credentials grant pattern and set it up with global setting. 

Now setup all your proxy services with routes too and add oauth and client credential grant pattern and set to global too. Now any time a client has generated a token from the global token url they can go call any of your other oauth2 protected urls as well. If you want to limit different customers to certain proxies you will use the ACL plugin to do so.

**Comment by lukasa1993** [comment]
> “ how i enable oauth2 on two routes so i can use the same token.”
> 
> I do this in Kong already. I don’t use the scopes field as its not really required for Oauth2 client credential grant pattern. Maybe it would help to post your Kong resources to see where you are going wrong. Setup a token gen route. Make it client credentials grant pattern and set it up with global setting.
> 
> Now setup all your proxy services with routes too and add oauth and client credential grant pattern and set to global too. Now any time a client has generated a token from the global token url they can go call any of your other oauth2 protected urls as well. If you want to limit different customers to certain proxies you will use the ACL plugin to do so.

what i give to ouath2 plugin as service.id ? 

@jeremyjpj0916 also how would i allow to have public routes after i enable it globally ?
